src/main/kotlin/no/nav/familie/ba/mottak/integrasjoner/InfotrygdBarnetrygdClient.kt

```diff
@@ -0,0 +1,114 @@
+package no.nav.familie.ba.mottak.integrasjoner
+
+import io.swagger.annotations.ApiModelProperty
+import no.nav.commons.foedselsnummer.FoedselsNr
+import no.nav.familie.ba.mottak.integrasjoner.Sakspart.*
+import no.nav.familie.http.client.AbstractRestClient
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.core.env.Environment
+import org.springframework.stereotype.Component
+import org.springframework.web.client.RestOperations
+import java.net.URI
+import java.time.LocalDate
+
+@Component
+class InfotrygdBarnetrygdClient(@Value("\${FAMILIE_BA_INFOTRYGD_BARNETRYGD_API_URL}/infotrygd/barnetrygd")
+                                private val clientUri: URI,
+                                @Qualifier("jwtBearer") restOperations: RestOperations,
+                                private val environment: Environment)
+    : AbstractRestClient(restOperations, "infotrygd_barnetrygd_replika") {
+
+    fun hentLøpendeUtbetalinger(søkersIdenter: List<String>, barnasIdenter: List<String>): InfotrygdSøkResponse<StønadDto> {
+        return infotrygdResponseFra(
+                postRequest = { postForEntity(uri("stonad"), mapTilInfotrygdSøkRequest(søkersIdenter, barnasIdenter)) },
+                onFailure = { ex -> IntegrasjonException("Feil ved søk etter stønad i infotrygd.", ex, uri("stonad")) }
+        )
+    }
+
+    fun hentSaker(søkersIdenter: List<String>, barnasIdenter: List<String>): InfotrygdSøkResponse<SakDto> {
+        return infotrygdResponseFra(
+                postRequest = { postForEntity(uri("saker"), mapTilInfotrygdSøkRequest(søkersIdenter, barnasIdenter)) },
+                onFailure = { ex -> IntegrasjonException("Feil ved uthenting av saker fra infotrygd.", ex, uri("saker")) }
+        )
+    }
+
+    private fun <T> infotrygdResponseFra(postRequest: () -> InfotrygdSøkResponse<T>,
+                                         onFailure: (Throwable) -> RuntimeException): InfotrygdSøkResponse<T> {
+        return if (environment.activeProfiles.contains("e2e"))  InfotrygdSøkResponse(emptyList(), emptyList())
+        else runCatching(postRequest).getOrElse { throw onFailure(it) }
+    }
+
+    private fun uri(endepunkt: String) = URI.create("$clientUri/$endepunkt")
+
+    private fun mapTilInfotrygdSøkRequest(søkersIdenter: List<String>,
+                                          barnasIdenter: List<String>): InfotrygdSøkRequest {
+        return InfotrygdSøkRequest(søkersIdenter.map { FoedselsNr(it) }, barnasIdenter.map { FoedselsNr(it) })
+    }
+}
+
+data class InfotrygdSøkRequest(val brukere: List<FoedselsNr>,
```

Comment:
Jeg er skeptisk til å bruke en pakke administrert av et annet team for FoedselsNr. Ser at dette kan bety problemer når vi skal ta ibruk den nye testident til dolly/pdl

Reply:
skal teste å bruke liste av String istedet 👍